main.go

```diff
@@ -158,38 +170,220 @@ func main() {
 		),
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(
-			err, "unable to create controller", "controller",
+			err, "Unable to create controller", "controller",
 			"AerospikeCluster",
 		)
 		os.Exit(1)
 	}
 
 	if err = (&asdbv1.AerospikeCluster{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "v1-webhook", "AerospikeCluster")
+		setupLog.Error(err, "Unable to create webhook", "v1-webhook", "AerospikeCluster")
 		os.Exit(1)
 	}
 	// +kubebuilder:scaffold:builder
 
 	if err := mgr.AddHealthzCheck("health", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up health check")
+		setupLog.Error(err, "Unable to set up health check")
 		os.Exit(1)
 	}
 
 	if err := mgr.AddReadyzCheck("check", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up ready check")
+		setupLog.Error(err, "Unable to set up ready check")
 		os.Exit(1)
 	}
 
+	go migrateAerospikeClusters(client, &options, watchNs)
```

Comment:
We can only migrate clusters after manager, webhooks are all setup. which is the part of `mgr.Start()` call. And this is the blocking call which means code written after this will not get executed. Hence wrote it as a routine which starts before `mgr.Start()` and wait for the controller to be ready internally.
2- No, there is no dependency between reconciler and this go routine.